nevergrad/functions/photonics/photonics.py

```diff
@@ -220,3 +227,240 @@ def morpho(X: np.ndarray) -> float:
.60170056862522e-08,9.409358023797568e-07,1.248433292295764e-06,2.432542842223933e-06,9.294665645046552e-07,9.525408765264329e-07,2.162828303733955e-06,9.796725833977744e-07,1.41603257182757e-06,1.945997785373929e-06,1.067556720928292e-06,2.295222826792951e-06,1.389476379674632e-06,2.038935088503858e-06,2.686898391984566e-06,2.606573997363201e-06,3.420484388282711e-06,1.221179903790976e-06,1.941717334337739e-06,1.998725058971436e-06,3.621037917249162e-06,2.756177464057926e-06,2.396955749673525e-06,3.191593836753927e-06,2.222898416496129e-06,3.135775905963728e-06,3.094545696973094e-06,2.308754138086146e-06,3.486689491160035e-06,2.735948676632181e-06,3.422533593847961e-06,3.058280468577859e-06,3.141325971743081e-06,3.389681161298935e-06,2.924405338201114e-06,3.447472699018933e-06,3.35444817597395e-06,3.328796513451797e-06,3.3770175856198e-06,3.24924624081506e-06,3.431650467587238e-06,3.239779522326613e-06,3.158921956265105e-06,2.688218794534419e-06,2.571136935934535e-06,2.720798428603173e-06,3.008656800449967e-06,2.589779680073297e-06,3.090713489548284e-06,3.111110242665669e-06,2.393841407589194e-06,2.485404062127128e-06,2.933022778401922e-06,2.643061276644058e-06,2.729091196791937e-06,2.7958345780385e-06,2.971122882188491e-06,3.003372536255901e-06,2.751959784016386e-06,2.753997891058559e-06,3.381243871037929e-06,2.98505032228905e-06,2.906256840247643e-06,3.010825435774416e-06,3.360697127010841e-06,3.107052043575303e-06,2.971585320650037e-06,3.226983246915034e-06,3.212559188118814e-06,3.022115368162559e-06,2.331823906147389e-06,1.427133748899321e-06,2.465707078932161e-06,2.797071500393035e-06,2.876304339184951e-06,3.010895806844652e-06,3.236956637109175e-06,3.209150011356218e-06,2.79396793513546e-06,2.756323112067292e-06,3.104266555555191e-06,3.204903619918021e-06,2.781041453137927e-06,2.578405624097193e-06,2.661776161182822e-06,3.111575093849624e-06,2.801864373456658e-06,2.715455454830512e-06,2.680213542432618e-06,2.588972021247196e-06,2.766476652201237e-06,2.748225734181544e-06,2.910707348001561e-06,2.682509448864094e-06,2.364207064188046e-06,2.706963074074522e-06,3.01308495030957e-06,2.847426183198532e-06,2.526428465896353e-06,2.379955506526093e-06,2.966208970884064e-06,2.924132700569002e-06,2.954722401619664e-06,2.372292096977473e-06,2.771920799466877e-06,2.715075491263283e-06,2.733251775736485e-06,2.637931346260028e-06,2.481632977003321e-06,2.517454545959753e-06,2.290859659590205e-06,2.107332069264829e-06,2.038904849049677e-06,2.115801086825502e-06,2.152968874461554e-06,2.341870401830494e-06,2.485832722474962e-06,2.48949077155395e-06,2.243753265174727e-06,2.199865764147931e-06,2.190001750702956e-06,2.162933257158706e-06,2.228332993296218e-06,2.327342877455008e-06,2.345700035684736e-06,2.392619122417268e-06,2.423130490413747e-06,2.4555244654932e-06,2.467641760607109e-06,2.489093154901101e-06,2.452289607924586e-06,2.400743844490309e-06,2.364552887733202e-06,2.382105724736161e-06,2.365712683394759e-06,2.316531186860184e-06,2.285427173816204e-06])
```

Comment:
do you expect to modify this? If not I'll update the PR after dumping it into a file

Reply:
Side node: I should do this at a moment where you don't touch anything in the code, because my editor will automatically reformat a lot of stuffs (typically the spaces after `,` or around `=`)

Reply:
We're not touching anything right now, go ahead...

Reply:
Do you mean we can probably find a module for atom which would be able to help us put everything in the right format (pep8) ?

Reply:
> Do you mean we can probably find a module for atom which would be able to help us put everything in the right format (pep8) ?
Many of them available yes, the most basic one being autopep8 for instance. I am not familiar with atom but I guess this can work ?
https://atom.io/packages/python-autopep8

Reply:
there are probably other useful packages to help writting compliant code (autopep8 is really basic). The main aims are used are pylint and/or flake8.
If you want to entirely stop thinking about code style in python, then `black` it's just a formatter that don't leave you anything to say... it's a bit violent but fun lol

Reply:
What is black ? A python script ? autopep8 seems to work, it's a basic script indeed, but there is no package for that for atom. Sigh.

Reply:
I feel like an intern.

Reply:
Black : https://github.com/psf/black with moto "the uncompromising code formatter"
While pep8 provides rough guidance for style but leaves some flexibility. autopep8 implements a few of the rules in pep8 (basically the main ones which you should never avoid, like some spacing rules around operators). Black on the other hand implements a lot more of pep8, but does not leave any flexibility. It makes your code waaaay more compliant but the lack of flexibility can make the code a bit ugly :D 

https://atom.io/packages/python-black

Reply:
I've edited the code there to use pep8 and data files: https://github.com/facebookresearch/nevergrad/pull/926
I wanted to make a PR on your repo, but it seems I just cannot, not sure why :s (is or was the repo private?)
I'll try more on Monday I guess

Reply:
I just removed the restricted options for the PR, I saw the PR #926. What have we do next? Close this PR?

Reply:
What was the restriction? (I am not familiar about what it is to be on the other hand of the PR ;) )
I still can't do a PR on your PR, and had no time to investigate further so far, I put this in my todolist for tomorrow. I'd rather we do not close this PR but merge my fixes in it, if possible at all, I'll investigate this

Reply:
> > For now I still don't know how I could create a PR on your branch that you could merge inside yourself. I also tried to push the update directly into your branch in case it worked but I cannot.
> > I'm considering updating the PR myself on my own branch until it can be merged into master, and then we'll need to move to the next steps in yet another PR
> 
> Ok, I have a branch here with some update:
> [jrapin#6](https://github.com/jrapin/nevergrad/pull/6)
> Compared to your branch, here are the differences:
> 
> * the data is dumped into specific files, which makes the code more readable (and maybe faster, although if speed matters we can find other ways to speed things up)
> * I used autopep8 to add spaces
> * I commented a few variables which were not used (`R`, and `r`)
> * I updated some names which were undefinied `pi` -> `np.pi`, `sin` -> `np.sin`, `espSia` -> `espcSi` (although I am unsure for this one, you may need to confirm
> * I updated the tests so that they are able to find the functions to use.
> 
> However there is an issue somewhere that I can solve myself, because I get this in the `test_photosic_reference` test:
> 
> ```
>        return interp_func(x, xp, fp, left, right)
> E       ValueError: fp and xp are not of the same length.
> ```
**_>Zut, I will work on this in priority_** 

> Another remark is that a lot of code is shared between the `cf_photosic_realist` and `cf_photosic_reference` functions. Is there a way to factorize some code? It may be helpful on the long run if we need to update them (-> updating in only one place instead of 2 places). Then again it may not be convenient so up to you.

**_> I will create a function _absorption(X)_ which will be called by the two other functions `cf_photosic_realist` and `cf_photosic_reference`, in a way that the two last functions will be more compact._** 

> From this, I am not sure how to proceed, since the code is still not working I would rather not merge it into nevergrad yet.
> I see two options:
> 
> * you solve it directly in your code, the PR here [jrapin#6](https://github.com/jrapin/nevergrad/pull/6) may help a bit to provide some of the solutions (in particular, please use the datafiles that I provide and remove the huge array definitions)
> * copy paste the code from the PR I provide, and then work from there. It may be simpler, then again because it is reformatted you may find it more difficult, so up to you.

**_>I chose the second option, I will learn to deal whit the format !_** 

> If you have other options in mind, I am all ears ;)
> Also, if _anything_ is unclear in my comment, please ask questions and I will clarify!

>
>For the last remark, 
>